abi/abit/types.go

```diff
@@ -108,11 +130,24 @@ var (
 )
 
 func List(et Type) Type {
-	return Type{
+	t := Type{
 		Name: et.Signature(),
 		Kind: L,
 		Elem: &et,
 	}
+	t.Name = t.Signature()
```

Comment:
What's the purpose of `Name`? is it supposed to be the signature? Previously it was just returning the underlying type

Reply:
Great question! I think I answered it. I thought it was easier to just push a commit with the docs and a few style bytes.